model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/functional_linear.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from torch import nn
+import torch.nn.functional as F
+
+from model_compression_toolkit.core.common.graph.graph_matchers import NodeOperationMatcher
+from model_compression_toolkit.core import common
+from model_compression_toolkit.core.common import BaseNode, Graph
+from model_compression_toolkit.core.common.graph.functional_node import FunctionalNode
+from model_compression_toolkit.core.pytorch.constants import *
+from model_compression_toolkit.logger import Logger
+
+
+class FunctionalLinear(common.BaseSubstitution):
+    """
+    Replace functional layer_norm with LayerNorm.
+    """
+
+    def __init__(self):
+        """
+        Matches: functional linear
+        """
+        func_node = NodeOperationMatcher(F.linear)
+        super().__init__(matcher_instance=func_node)
+
+    def substitute(self,
+                   graph: Graph,
+                   func_node: FunctionalNode) -> Graph:
+        """
+        Substitute functional.linear and its inputs with Linear.
+        Args:
+            graph: Graph we apply the substitution on.
+            node: node that match the pattern in the substitution init.
+
+        Returns:
+            Graph after applying the substitution.
+        """
+
+        # Create new node of layer convolution
+        if 1 not in func_node.weights:
+            Logger.critical(f'Weight input missing for node {func_node.name}.')  # pragma: no cover
+
+        weight = func_node.weights[1]
+        bias = func_node.weights.get(2, None)
```

Comment:
make sure the bias positional weight is saved as 2 even when the linear is called with bias as kwarg